pyocd/probe/stlink/usb.py

```diff
@@ -126,8 +128,12 @@ def __init__(self, dev):
         # reading the strings, everything is ok. This workaround doesn't cause any issues with
         # Linux or macOS.
         try:
-            self._serial_number = self._dev.serial_number
-            self._vendor_name =  self._dev.manufacturer
+            if self._dev.idProduct == 0x3748:  # Workaround for unprintable characters in the ST-Link V2 probes
```

Comment:
Do you know if it's only the V2 probe firmware (PID==0x3748) that has this issue? Or does it affect V2-1 and V3?

Reply:
I'm not 100% sure, but we are using V2-1 without problems. On the other hand, we are using V2 as the boxed ready-to-use probe [here](https://www.st.com/en/development-tools/st-link-v2.html), and V2-1 as cuttable PCB as part of the Nucleos boards.

Reply:
I'll try to get an answer from ST.

In the meantime, this **if** statement needs an **else** clause that sets `self._serial_number` for other PIDs. Or you could leave original line 129 in place and just override for 0x3748. Sorry I didn't notice this before (but that's what CI is for! 😄 ).

Reply:
Based on the info from @schstm, could you please change this to check for a serial number size of 26 characters?

Also maybe remove the hex conversion, so each UTF-16 char is split into two 8-bit ASCII chars, depending on what @schstm says.

Reply:
> In the meantime, this if statement needs an else clause that sets self._serial_number for other PIDs. Or you could leave original line 129 in place and just override for 0x3748. Sorry I didn't notice this before (but that's what CI is for! 😄 ).

Sorry for the typo, lost the **else** in the copy-paste.